### PR TITLE
Feature/signin signup

### PR DESCRIPTION
**Pull request by mbalci85**

### CODE REVIEW

File: src/pages/Auth.tsx [hunk 1]
```diff
@@ -1,12 +1,55 @@
...
 
+	const [username, setUsername] = useState<string>('');
+	const [email, setEmail] = useState<string>('');
+	const [password, setPassword] = useState<string>('');
+
+	const signin = async (e: FormEvent) => {
+		e.preventDefault();
+		try {
+			const response = await axios.post(`${base_url}/users/signin`, { email, password });
+			console.log(response.data);
```
Comment (mrselifbalci) [inline src/pages/Auth.tsx]: Please remove this console.log

**Review by mrselifbalci — CHANGES_REQUESTED** [review]

**Review by mrselifbalci — CHANGES_REQUESTED** [review]
You should add the .env file to .gitignore

File: src/pages/Auth.tsx (hunk 1 above)
Comment (mbalci85) [reply src/pages/Auth.tsx]: @mrselifbalci removed!

**Comment by mbalci85** [comment]
@mrselifbalci I have created enums, theme, and textfield generator component. Please revise new updates.

File: src/components/TextFieldGenerator.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { TextField, TextFieldProps } from '@mui/material';
+import { ChangeEvent } from 'react';
+
+interface TextFieldGeneratorProps {
```
Comment (mrselifbalci) [inline src/components/TextFieldGenerator.tsx]: We should probably name this component something like StyledTextField or CustomTextField. Textfield generator sounds like a function name.

**Review by mrselifbalci — CHANGES_REQUESTED** [review]

File: src/components/TextFieldGenerator.tsx (hunk 2 above)
Comment (mbalci85) [reply src/components/TextFieldGenerator.tsx]: @mrselifbalci Oki doki ma'am

Comment (mbalci85) [reply src/components/TextFieldGenerator.tsx]: @mrselifbalci done

**Review by mrselifbalci — APPROVED** [review]

**Pull request merged**